common/lib/xmodule/xmodule/js/src/capa/display.js

```diff
@@ -377,406 +441,461 @@
+            this.focus_on_notification('hint');
         };
 
-        Problem.prototype.focus_on_save_notification = function () {
-            return this.focus_on_notification('save');
+        Problem.prototype.focus_on_save_notification = function() {
+            this.focus_on_notification('save');
         };
 
-        /*
-         # 'submit_fd' uses FormData to allow file submissions in the 'problem_check' dispatch,
-         #      in addition to simple querystring-based answers
-         #
-         # NOTE: The dispatch 'problem_check' is being singled out for the use of FormData;
-         #       maybe preferable to consolidate all dispatches to use FormData
-         */
-
+        Problem.prototype.submit_fd = function() {
+            /**
+             * 'submit_fd' uses FormData to allow file submissions in the 'problem_check' dispatch,
+             *      in addition to simple querystring-based answers
+             *
+             * NOTE: The dispatch 'problem_check' is being singled out for the use of FormData;
+             *       maybe preferable to consolidate all dispatches to use FormData
+             */
+            var abortSubmission, error, errorHtml, errors, fd, fileNotSelected, fileTooLarge, maxFileSize,
+                requiredFilesNotSubmitted, settings, timeoutId, unallowedFileSubmitted, i, len,
+                that = this;
 
-        Problem.prototype.submit_fd = function () {
-            var abort_submission, error, error_html, errors, fd, file_not_selected, file_too_large, max_filesize, required_files_not_submitted, settings, timeout_id, unallowed_file_submitted, _i, _len,
-                _this = this;
+            // If there are no file inputs in the problem, we can fall back on submit.
             if (this.el.find('input:file').length === 0) {
                 this.submit();
                 return;
             }
             this.enableSubmitButton(false);
             if (!window.FormData) {
-                alert("Submission aborted! Sorry, your browser does not support file uploads. If you can, please use Chrome or Safari which have been verified to support file uploads.");
+                alert(gettext('Submission aborted! Sorry, your browser does not support file uploads. If you can, please use Chrome or Safari which have been verified to support file uploads.')); // eslint-disable-line max-len, no-alert
                 this.enableSubmitButton(true);
                 return;
             }
-            timeout_id = this.enableSubmitButtonAfterTimeout();
+            timeoutId = this.enableSubmitButtonAfterTimeout();
             fd = new FormData();
-            max_filesize = 4 * 1000 * 1000;
-            file_too_large = false;
-            file_not_selected = false;
-            required_files_not_submitted = false;
-            unallowed_file_submitted = false;
+
+            // Sanity checks on submission
+            maxFileSize = 4 * 1000 * 1000;
```

Comment:
Somewhat unrelated question here, why are we using 1000 and not 1024 here I wonder? Also why not include the expected unit in the variable name (GB, MB, etc..)? This should in no way impact the merge of this PR.